Headless mode

cmontesano opened the pull request:
The runner dialog is still created, but never shown. The big exceptions are with the progress functions, where we have to keep track of whether or not we're headless and route things accordingly.

A simple working example tool:

```python
import time
import xappt


@xappt.register_plugin
class HeadlessTest(xappt.BaseTool):
    headless = True

    @classmethod
    def name(cls) -> str:
        return "headless-test"

    @classmethod
    def help(cls) -> str:
        return "Just a test to work out issues with headless mode."

    @classmethod
    def collection(cls) -> str:
        return "System"

    def execute(self, **kwargs) -> int:
        if not self.interface.ask("Continue?"):
            return 0
        self.interface.progress_start()
        for i in range(100):
            time.sleep(0.1)
            self.interface.progress_update(message=f"Message {i + 1:03d}", percent_complete=(i / 100))
        self.interface.progress_end()
        self.interface.message("Complete")
        return 0
```

cmontesano committed:
relocating code to center windows for reuse

cmontesano committed:
initial support for headless mode

cmontesano committed:
bump revision

kennedy0 commented:
Looks great. The only thing I noticed is that the python process is still running after the last dialog closes. I tried explicitly calling `self.close()` in `xappt_qt/interface.py`, but no dice.

```python
    def invoke(self, plugin: BaseTool, **kwargs):
        self.headless = kwargs.get('headless')
        if self.headless:
            center_widget(self.runner)
            self.instance.exec_()
            plugin.execute()
            self.close()  # <-- This didn't work
            return
```

cmontesano commented:
Good catch. I'll see what I can do. I seem to remember something about an application auto-quitting when the last window closes, although maybe I'm thinking of something else. In any case I think that kind of behavior can be implemented with just a little bookkeeping.

cmontesano committed:
support headless mode if launching in same process

cmontesano committed:
detect if we're running from the browser or launcher

cmontesano committed:
get rid of internal application

This was never used when running from the browser or the launcher. This was only really used when invoking from the xappt core itself.

cmontesano commented:
Ok, so I'm detecting if we're running from the launcher, and if so, plus if we're headless, then I quit after execution. This seems to do the trick.
